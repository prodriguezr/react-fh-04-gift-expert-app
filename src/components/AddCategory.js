import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handlerOnChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handlerOnSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length === 0) return;

    setCategories((oldCategories) => [inputValue, ...oldCategories]);

    setInputValue('');
  };

  return (
    <form onSubmit={handlerOnSubmit}>
      <input type='text' value={inputValue} onChange={handlerOnChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
