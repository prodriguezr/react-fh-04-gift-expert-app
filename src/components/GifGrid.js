import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p className='animate__flash'>Loading...</p>}

      <div className='card-grid'>
        {images.map(({ id, title, url }) => (
          <GifGridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
