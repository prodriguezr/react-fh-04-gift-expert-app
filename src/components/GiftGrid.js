import { GiftGridItem } from './GiftGridItem';

import { useFetchGifts } from '../hooks';

export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifts(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p className='animate__flash'>Loading...</p>}

      <div className='card-grid'>
        {images.map(({ id, title, url }) => (
          <GiftGridItem key={id} title={title} url={url} />
        ))}
      </div>
    </>
  );
};
