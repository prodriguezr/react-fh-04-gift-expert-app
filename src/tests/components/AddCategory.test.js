import { shallow } from 'enzyme';

import { AddCategory } from '../../components';

describe('Tests on AddCategory component', () => {
  const setCategories = jest.fn();

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('It should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Must change the text box', () => {
    const input = wrapper.find('input');
    const value = 'Hello, World!';
    input.simulate('change', { target: { value } });
  });

  test('You should not post the information with submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('You must call setCategories and clear the text box', () => {
    const input = wrapper.find('input');

    const value = 'Hello, World!';
    input.simulate('change', { target: { value } });

    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(input.prop('value')).toBe('');
  });
});
