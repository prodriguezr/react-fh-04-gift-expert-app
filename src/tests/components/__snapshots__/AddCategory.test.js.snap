// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AddCategory component It should display correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
